### PR TITLE
Updated code of conduct to contributor covenant v2.1

### LINKED ISSUE #4160: Update Code of Conduct
PyBaMM's code of conduct is adapted from contributor covenant 1.4. It would be good to update to the newest version (v2.1) as done in https://github.com/pybamm-team/pybamm-cookiecutter/pull/10. 

This probably needs formal approval from the Steering Council at some point, but given than it is an update on our current CoC it should be fine to do it retrospectively (tagging SC members @valentinsulzer @martinjrobins @Scottmar93 @Saransh-cpp @gjo97).

### PR DESCRIPTION
**Pull request by santacodes**
# Description

Fixes #4160 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Review by brosaplanella — APPROVED** [review]

File: CODE-OF-CONDUCT.md [hunk 1]
```diff
@@ -1,46 +1,135 @@
+[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.1-4baaaa.svg)](code_of_conduct.md)
```
Comment (arjxn-py) [inline CODE-OF-CONDUCT.md:1]:
```suggestion
[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.1-4baaaa.svg)](https://www.contributor-covenant.org/version/2/1/code_of_conduct.html)
```
Looks like linkchecker is failing becauseof this.

Comment (kratman) [inline CODE-OF-CONDUCT.md:1]: The link in this badge points to this same file. Is the link supposed to point to something on contributor covenant, or is the badge supposed to be in a different file?

**Review by kratman — CHANGES_REQUESTED** [review]

**Review by arjxn-py — CHANGES_REQUESTED** [review]

File: CODE-OF-CONDUCT.md (hunk 1 above)
Comment (arjxn-py) [reply CODE-OF-CONDUCT.md:1]: I just saw your comment, this badge should be pointing to [contributor covenant's code of conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct.html) only.

**Review by arjxn-py — APPROVED** [review]

**Comment by kratman** [comment]
@santacodes This same issue is on pybamm-cookiecutter. Can you fix it there as well?

**Comment by santacodes** [comment]
> @santacodes This same issue is on pybamm-cookiecutter. Can you fix it there as well?

Thank you for noticing, yes I did fix it!

**Comment by agriyakhetarpal** [comment]
We should adapt Lychee to the cookiecutter repository as well, @santacodes.

**Review by kratman — APPROVED** [review]

**Pull request merged**